contrib/openapi/spec.yml

```diff
@@ -1,453 +1,282 @@
-        $ref: '#/definitions/Syntax'
+        type: string
+        enum:
+          - actionscript
+          - ada
+          - apacheconf
+          - applescript
+          - asymptote
+          - bash
+          - bat
+          - boo
+          - brainfuck
+          - bro
+          - c
+          - clojure
+          - cmake
+          - coffeescript
+          - common-lisp
+          - coq
+          - cpp
+          - csharp
+          - css
+          - cython
+          - d
+          - dart
+          - delphi
+          - diff
+          - django
+          - dylan
+          - elixir
+          - erb
+          - erlang
+          - factor
+          - fancy
+          - felix
+          - fortran
+          - fsharp
+          - gas
+          - gherkin
+          - glsl
+          - gnuplot
+          - go
+          - groovy
+          - haskell
+          - html
+          - html+php
+          - ini
+          - io
+          - java
+          - javascript
+          - lighttpd
+          - llvm
+          - lua
+          - makefile
+          - matlab
+          - minid
+          - modelica
+          - modula2
+          - moonscript
+          - mupad
+          - nasm
+          - nemerle
+          - newlisp
+          - nginx
+          - nimrod
+          - objectivec
+          - ocaml
+          - octave
+          - opa
+          - perl
+          - postscript
+          - powershell
+          - prolog
+          - pycon
+          - python
+          - python3
+          - rbcon
+          - rebol
+          - restructuredtext
+          - ruby
+          - scala
+          - scheme
+          - scilab
+          - smalltalk
+          - smarty
+          - splus
+          - sql
+          - systemverilog
+          - tcl
+          - tex
+          - text
+          - vala
+          - vb.net
+          - verilog
+          - vhdl
+          - vim
+          - xml
+          - yaml
       tags:
         type: array
         items:
           type: string
-        description: List of string tags assigned to the snippet.
-      revision:
-        type: integer
-        minimum: 0
-        description: |
-          Current (latest) revision of the snippet.
-      changeset:
-        $ref: '#/definitions/Changeset'
-      title:
-        type: string
-        description: Title of the snippet.
-      is_public:
-        type: boolean
-        description: |
-          If `true`, snippet is visible for all users, otherwise it's
-          considered to be private and visible only to the author.
       created_at:
         type: string
         format: date-time
-        description: Date-time value when the snippet was created.
+        readOnly: true
       updated_at:
         type: string
         format: date-time
-        description: Date-time value when the snippet was updated last time.
-
+        readOnly: true
+    required:
+      - content
   Error:
     type: object
     properties:
-      code:
```

Comment:
I think it's useful to have a "machine-readable" `code` value to specify error details besides the HTTP status, but as long as we don't have such examples in the implementation right now, we might add this back later.